spec/models/gibct_site_mapper_spec.rb

```diff
@@ -3,17 +3,17 @@
 require 'rails_helper'
 
 RSpec.describe GibctSiteMapper, type: :model do
-  let(:preview_version) { Version.current_preview }
   let(:production_version) { Version.current_production }
   let(:sitemaps_path) { Rails.root.join('public', 'sitemap.xml.gz') }
   let(:preview_institution_fc) { '00000001' }
   let(:production_institution_fc) { '00000002' }
 
   before do
+    create :version, :production
+    version = create :version, :preview
     File.delete(sitemaps_path) if File.exist?(sitemaps_path)
-
     %i[preview production].each { |p| create :version, p }
-    create :institution, version: preview_version.number, facility_code: preview_institution_fc
+    create :institution, version: version.number, facility_code: preview_institution_fc
```

Comment:
Now it looks like we are creating unused `versions`.  2 on lines 12, 13 and 2 on line 15.
I think if you undo all these changes and just swap the creation order of `preview` and `production` on line 15, everything will work.
Remember, your `current_preview` has to be newer than your `current_production`. So change the creation order and that state will be achieved.

Reply:
Changes are in.